/*
 * This file was automatically generated by EvoSuite
 * Tue Nov 15 05:41:52 GMT 2022
 */

package tutorial;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.NoSuchElementException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import tutorial.LinkedListIterator;
import tutorial.Node;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class LinkedListIterator_ESTest extends LinkedListIterator_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Node<String> node0 = new Node<String>("{*3dV@#f!a");
      LinkedListIterator<String> linkedListIterator0 = new LinkedListIterator<String>(node0);
      String string0 = linkedListIterator0.next();
      assertEquals("{*3dV@#f!a", string0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Node<String> node0 = new Node<String>("");
      LinkedListIterator<String> linkedListIterator0 = new LinkedListIterator<String>(node0);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      LinkedListIterator<String> linkedListIterator0 = new LinkedListIterator<String>((Node<String>) null);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Node<Integer> node0 = new Node<Integer>((Integer) null);
      LinkedListIterator<Integer> linkedListIterator0 = new LinkedListIterator<Integer>(node0);
      linkedListIterator0.next();
      // Undeclared exception!
      try { 
        linkedListIterator0.next();
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("tutorial.LinkedListIterator", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>((Node<Object>) null);
      // Undeclared exception!
      try { 
        linkedListIterator0.remove();
        fail("Expecting exception: UnsupportedOperationException");
      
      } catch(UnsupportedOperationException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("tutorial.LinkedListIterator", e);
      }
  }
}
